packages/wrangler/src/index.tsx

```diff
@@ -1092,7 +1105,15 @@ export async function main(argv: string[]): Promise<void> {
       if (!shortScriptName) {
         throw new Error("Missing script name");
       }
-      const scriptName = `${shortScriptName}${args.env ? `-${args.env}` : ""}`;
+
+      const isLegacyEnv =
+        (args["legacy-env"] as boolean | undefined) ??
+        config.legacy_env ??
+        false;
```

Comment:
How about moving this logic, which is repeated below, into a small helper function?